File: 637. Average of Levels in Binary Tree/main.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public List<Double> averageOfLevels(TreeNode root) {
        Queue<TreeNode> listTreeNode = new LinkedList<TreeNode>();
        List<Double> result = new ArrayList<>();
        if (root == null) return result;
        listTreeNode.add(root);
            
        while (!listTreeNode.isEmpty()){
            int breadth = listTreeNode.size();
            double sum = 0.0;
            // use stream is much slower
            // result.add(listTreeNode.stream().mapToDouble(o->o.val).sum()/breadth);
            for (int i=0;i<breadth;++i ){
                TreeNode node = listTreeNode.poll();
                sum += node.val;
                if (node.left !=null){
                    listTreeNode.offer(node.left);
                }
                if (node.right != null){
                    listTreeNode.offer(node.right);
                }
            }
            result.add(sum/breadth);
            
        }
        return result;
    }
}

File: 637. Average of Levels in Binary Tree/main.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def averageOfLevels(self, root: TreeNode) -> List[float]:
        if not root:
            return [0]
        stack = [root]
        result = []
        while stack:
            count = len(stack)
            print(count)
            result.append(sum(x.val for x in stack[0:count])/count)
            new_stack = []
            for i in range(count):
                node = stack.pop()
                if node.left:
                    new_stack.append(node.left)
                if node.right:
                    new_stack.append(node.right)
            stack = new_stack
        return result
                